### PR TITLE
Move non transformer code out of AssessmentDto

### PR DESCRIPTION
**Pull request by nickmcmahon01**

### CODE REVIEW

File: src/main/java/uk/gov/justice/digital/oasys/service/domain/SectionHeader.java [hunk 1]
```diff
@@ -0,0 +1,32 @@
+package uk.gov.justice.digital.oasys.service.domain;
+
+public enum SectionHeader {
```
Comment (SWhitlam) [inline src/main/java/uk/gov/justice/digital/oasys/service/domain/SectionHeader.java]: Don't forget to make the required changes in the sentence plan API for this.

File: src/test/java/uk/gov/justice/digital/oasys/controller/AssessmentsControllerTest.java [hunk 2]
```diff
@@ -1,41 +1,22 @@
...
-import uk.gov.justice.digital.oasys.service.OffenderService;
-
-import java.util.Optional;
 
 import static io.restassured.RestAssured.given;
 import static org.assertj.core.api.Assertions.assertThat;
 import static org.mockito.ArgumentMatchers.eq;
 
-
+@Ignore
```
Comment (SWhitlam) [inline src/test/java/uk/gov/justice/digital/oasys/controller/AssessmentsControllerTest.java]: Did you mean to ignore this?

Comment (nickmcmahon01) [reply src/test/java/uk/gov/justice/digital/oasys/controller/AssessmentsControllerTest.java]: Yes, can I sort it as part of 547

File: src/main/java/uk/gov/justice/digital/oasys/api/AnswerDto.java [hunk 3]
```diff
@@ -23,18 +23,18 @@
     private Long ovpScore;
     private Long qaRawScore;
 
-    public static AnswerDto from(OasysQuestion question) {
-        if (question.getOasysAnswer() == null) {
-            return new AnswerDto(null, null, null,null,null, question.getFreeFormatAnswer(), null, null, null);
+    public static AnswerDto from(OasysQuestion question, OasysAnswer oasysAnswer) {
```
[the remark this answers is not in the document]
Comment (nickmcmahon01) [reply src/main/java/uk/gov/justice/digital/oasys/api/AnswerDto.java]: sorted, this was a bad merge

File: build.gradle [hunk 4]
```diff
@@ -119,6 +146,8 @@ dependencies {
     testImplementation 'com.github.tomakehurst:wiremock:2.25.0'
     testImplementation 'com.revinate:assertj-json:1.2.0'
     testImplementation('org.assertj:assertj-core')
+    testImplementation "com.google.guava:guava:28.2-jre"
+    testImplementation "com.google.code.gson:gson:2.8.6"
```
[the remark this answers is not in the document]
Comment (nickmcmahon01) [reply build.gradle]: removed.

File: build.gradle [hunk 5]
```diff
@@ -88,12 +110,17 @@ dependencies {
     implementation 'com.fasterxml.jackson.datatype:jackson-datatype-jsr310:2.10.0'
     implementation 'com.github.ben-manes.caffeine:caffeine:2.8.1'
     implementation 'com.github.timpeeters:spring-boot-graceful-shutdown:2.1.1'
+    implementation "com.querydsl:querydsl-core:${queryDslVersion}"
+    implementation "com.querydsl:querydsl-jpa:${queryDslVersion}"
+    implementation "com.google.guava:guava:28.2-jre"
```
[the remark this answers is not in the document]
Comment (nickmcmahon01) [reply build.gradle]: removed

File: src/main/java/uk/gov/justice/digital/oasys/api/AnswerDto.java [hunk 6]
```diff
@@ -43,6 +43,13 @@ public static AnswerDto from(OasysQuestion question) {
                 refAnswer.map(RefAnswer::getQaRawScore).orElse(null));
     }
 
+    private static AnswerDto from(OasysQuestion question) {
```
[the remark this answers is not in the document]
Comment (nickmcmahon01) [reply src/main/java/uk/gov/justice/digital/oasys/api/AnswerDto.java]: combined.

File: src/main/java/uk/gov/justice/digital/oasys/api/simple/AssessmentDto.java [hunk 7]
```diff
@@ -0,0 +1,76 @@
...
+    @JsonProperty("sections")
+    private Collection<SectionDto> sections;
+    //TODO: we don't need these until we get to update the sentence plan code
+    //@JsonProperty("oasysBcsParts")
+    //private Collection<OasysBcsPartDto> oasysBcsParts;
+    //@JsonProperty("qaReview")
+    //private QaReviewDto qaReview;
+    //TODO: why specifically childsafeguarding?
+    @JsonProperty("childSafeguardingIndicated")
+    private Boolean childSafeguardingIndicated;
```
[the remark this answers is not in the document]
Comment (nickmcmahon01) [reply src/main/java/uk/gov/justice/digital/oasys/api/simple/AssessmentDto.java]: ta

File: src/main/java/uk/gov/justice/digital/oasys/controller/SentencePlansController.java [hunk 8]
```diff
@@ -54,11 +52,10 @@ public SentencePlansController(SentencePlanService sentencePlanService) {
             @ApiResponse(code = 200, message = "OK")})
     public ResponseEntity<List<BasicSentencePlan>> getSentenceBasicPlansForOffenderBookingId(@PathVariable("identityType") String identityType,
```
[the remark this answers is not in the document]
Comment (nickmcmahon01) [reply src/main/java/uk/gov/justice/digital/oasys/controller/SentencePlansController.java]: I have renamed the method.

File: src/main/java/uk/gov/justice/digital/oasys/jpa/repository/simple/SimpleAssessmentRepository.java [hunk 9]
```diff
@@ -0,0 +1,72 @@
...
+    }
+
+    private JPAQuery<Assessment> getAssessmentsQueryForOffender(Long offenderId){
+        var query = queryFactory.selectFrom(QAssessment.assessment);
+        return query.where(QAssessment.assessment.group.offenderPk.eq(offenderId));
+    }
+
+    private void filterQuery(JPAQuery<Assessment> query, String filterGroupStatus, String filterAssessmentType, Boolean filterVoided, String filterAssessmentStatus) {
+
+        if (Optional.ofNullable(filterAssessmentStatus).isPresent()) {
```
[the remark this answers is not in the document]
Comment (nickmcmahon01) [reply src/main/java/uk/gov/justice/digital/oasys/jpa/repository/simple/SimpleAssessmentRepository.java]: I've changed it to Objects.NonNull.
I've changed the Optional<String> to required = false annotation at the controller level - I know Spring abuses it but the Optional<> people say it shouldn't be used in method signatures

File: src/main/java/uk/gov/justice/digital/oasys/controller/AssessmentsController.java [hunk 10]
```diff
@@ -30,81 +25,41 @@ public AssessmentsController(AssessmentsService assessmentsService) {
         this.assessmentsService = assessmentsService;
     }
 
-    @RequestMapping(path = "/assessments/oasysSetId/{oasysSetId}", method = RequestMethod.GET)
-    @ApiResponses({
-            @ApiResponse(code = 404, message = "Assessment not found"),
-            @ApiResponse(code = 200, message = "OK")})
+    @GetMapping(path = "/assessments/oasysSetId/{oasysSetId}")
+    @ApiOperation(value = "Gets an assessment by its identity")
```
[the remark this answers is not in the document]
Comment (nickmcmahon01) [reply src/main/java/uk/gov/justice/digital/oasys/controller/AssessmentsController.java]: I have added these back in.

**Pull request merged**